admin_guide/backup_restore.adoc

```diff
@@ -208,6 +208,194 @@ that {product-title} was previously installed.
+# iptables -A OS_FIREWALL_ALLOW -p tcp -m state \
+  --state NEW -m tcp --dport 2380 -j ACCEPT
+# iptables-save
+----
+
+. Generate the required certificates for the new host. On a surviving etcd host:
++
+.. Create a copy of the *_/etc/etcd/ca/_* directory.
+
+.. Set the variables and working directory for the certificates, ensuring to create the *_PREFIX_* directory if one has not been created:
++
+----
+# cd /etc/etcd
+# export NEW_ETCD="<NEW_HOST_NAME>"
+
+# export CN=$NEW_ETCD
+# export SAN="IP:<NEW_HOST_IP>"
+# export PREFIX="./generated_certs/etcd-$CN/"
+----
+
+.. Create the *_server.csr_* and *_server.crt_* certificates:
++
+----
+# openssl req -new -keyout ${PREFIX}server.key \
+  -config ca/openssl.cnf \
+  -out ${PREFIX}server.csr \
+  -reqexts etcd_v3_req -batch -nodes \
+  -subj /CN=$CN
+
+# openssl ca -name etcd_ca -config ca/openssl.cnf \
+  -out ${PREFIX}server.crt \
+  -in ${PREFIX}server.csr \
+  -extensions etcd_v3_ca_server -batch
+----
+
+.. Create the *_peer.csr_* and *_peer.crt_* certificates:
++
+----
+# openssl req -new -keyout ${PREFIX}peer.key \
+  -config ca/openssl.cnf \
+  -out ${PREFIX}peer.csr \
+  -reqexts etcd_v3_req -batch -nodes \
+  -subj /CN=$CN
+
+# openssl ca -name etcd_ca -config ca/openssl.cnf \
+  -out ${PREFIX}peer.crt \
+  -in ${PREFIX}peer.csr \
+  -extensions etcd_v3_ca_peer -batch
+----
+
+.. Copy the *_ca.crt_* and *_etcd.conf_* files, and archive the contents of the directory:
++
+----
+# cp ca.crt ${PREFIX}
+# cp etcd.conf ${PREFIX}
+# tar -czvf ${PREFIX}${CN}.tgz -C ${PREFIX} .
+----
+
+.. Transfer the files to the new etcd hosts:
++
+----
+# scp ${PREFIX}${CN}.tgz  $CN:/etc/etcd/
+----
+
+. While still on the surviving etcd host, add the new host to the cluster, take the copy of etcd, and transfer it to the new host:
+
+.. Add the new host to the cluster:
++
+----
+# export ETCD_CA_HOST="<SURVIVING_ETCD_HOSTNAME>"
+# export NEW_ETCD="<NEW_ETCD_HOSTNAME>"
+# export NEW_ETCD_IP="<NEW_HOST_IP>"
+
+# etcdctl -C https://${ETCD_CA_HOST}:2379 \
+  --ca-file=/etc/etcd/ca.crt     \
+  --cert-file=/etc/etcd/peer.crt     \
+  --key-file=/etc/etcd/peer.key member add ${NEW_ETCD} https://${NEW_ETCD_IP}:2380
+
+ETCD_NAME="<NEW_ETCD_HOSTNAME>"
+ETCD_INITIAL_CLUSTER="<NEW_ETCD_HOSTNAME>=https://<NEW_HOST_IP>:2380,<SURVIVING_ETCD_HOST>=https:/<SURVIVING_HOST_IP>:2380
+ETCD_INITIAL_CLUSTER_STATE="existing"
+----
+
+.. Create a backup of the surviving etcd host, and transfer the contents to the new
+host:
++
+[NOTE]
+====
+Skip this step if version is lower than `etcd-2.3.7-4` or if etcd database size
+is smaller than 700mb.
+====
++
+----
+# export NODE_ID="<NEW_NODE_ID>"
+# etcdctl backup --keep-cluster-id --node-id ${NODE_ID} \
+  --data-dir /var/lib/etcd --backup-dir /var/lib/etcd/$NEW_ETCD-backup
+# tar -cvf $NEW_ETCD-backup.tar.gz -C /var/lib/etcd/$NEW_ETCD-backup/ .
+# scp $NEW_ETCD-backup.tar.gz $NEW_ETCD:/var/lib/etcd/
+----
+
+. On the new host, extract the backup data and set the permissions:
```

Comment:
This step is wrong:

- If you apply the step 5b before ("Create a backup of the surviving etcd host, and transfer the contents to the new
 +host:"), the correct tar command is: `tar -xf /var/lib/etcd/<NEW_ETCD_HOSTNAME>-backup.tar.gz -C /var/lib/etcd/`, note the `-backup.tar.gz`.
- If you do not apply step 5b before, this steps completely break the whole restore. The admin will probably try to get some backup copy (maybe the one of the first node) which has a wrong cluster-id and/or node-id. If you un-tars that and starts etcd, it will report that the cluster id and/or node-id does not match.

  The correct way for the "without 5b" case is that no backup of `/var/lib/etcd` is used. And, also important in case of re-use of a node: `/var/lib/etcd` must be purged from old data first. Otherwise, etcd will also complain that the cluster-id does not match.